diff --git a/meson.build b/meson.build
index eaac74a..a2924c6 100644
--- a/meson.build
+++ b/meson.build
@@ -78,7 +78,6 @@ endif
 
 wayland_protocols = dependency('wayland-protocols')
 wayland_client = dependency('wayland-client')
-tllist = dependency('tllist', version: '>=1.0.1', fallback: 'tllist')
 
 wayland_protocols_datadir = wayland_protocols.get_variable('pkgdatadir')
 
@@ -134,9 +133,10 @@ executable(
     'log.c', 'log.h',
     'shm.c', 'shm.h',
     'stride.h',
+    'tllist.h',
     image_format_sources,
     wl_proto_src + wl_proto_headers, version,
-    dependencies: [pixman, png, jpg, webp, svg, wayland_client, tllist],
+    dependencies: [pixman, png, jpg, webp, svg, wayland_client],
     install: true)
 
 summary(
